plugins/inventory/nb_inventory.py

```diff
@@ -962,6 +964,12 @@ def extract_dns_name(self, host):
 
         return ip_address.get("dns_name")
 
+    def extract_serial(self, host):
+        return host.get("serial")
+
+    def extract_asset_tag(self, host):
+        return host.get("asset_tag")
+
```

Comment:
Shouldn't these be return a default value of `None` if they are not set? Such as 
```python
host.get("serial", None)
```
because we cannot guarantee that the asset tag and serial are set? I see you have changed _a lot_ of the test files and set the values to `None`, which just means that now all our test data has the `serial` and `asset_tag` attributes with `None` values but I see other instances in this file, like:

```python
       site = host.get("site", None)
        if not isinstance(site, dict):
            # Device has no site
            return []
```

Basically, you need to have this code deal with instances where `asset_tag` and `serial` are not attributes that get returned from the NetBox API?

Reply:
Thanks for the feedback. Re: test files I was/am really just fumbling around trying things to get a good CI.
I'll add the default handling and make the test files rational again and see what haps.